python/mp/rosary.py

```diff
@@ -206,22 +264,30 @@ def clear_effects(self):
             effect = self.effects[0]
             self.del_effect(effect.id)
 
+        # I know on the real rosary this is unneccessary, but it's
+        # annoying on the sim: @jdblair is sending 0,0,0 in the real
+        # thing wonky?
+        self.add_effect('set_color', 'all', 0, 0, 0)
+
     @dm.expose()
     def del_effect(self, id):
         """Delete an active effect by id."""
 
         effect = self.effect(id)
-        effect_paths = [effect.generate_osc_path(fn) for fn in\
-                        effect.dm.registered_methods.keys()]
-        self.effect_paths_to_unregister.extend(effect_paths)
-        self.effects.remove(effect)
+
+        if effect is not None:
+            effect_paths = [effect.generate_osc_path(fn) for fn in\
+                            effect.dm.registered_methods.keys()]
+            self.effect_paths_to_unregister.extend(effect_paths)
+            self.effects.remove(effect)
+
 
     def effect(self, id):
         """Return the Effect object of an active effect by specifying the Effect id."""
         for e in self.effects:
             if e.id == id:
                 return e
-        return 0
+        return None
```

Comment:
More pythonic I...think. It's less weird to get a Nonetype back than a completely different type (int).